### ISSUE TITLE
Vexl on F-Droid 🤖

### ISSUE THREAD

**Issue by tomas-bartos1** [vexl-it/vexl#4]
For app like this, I would say it is very important to be available through the FOSS channel without the need for Google or Apple intermediary. Third parties (especial non-transparent ones) are security risks.

- F-Droid: https://f-droid.org
- **Here is a guide how to submit new app into F-Droid repository: https://f-droid.org/docs/Submitting_to_F-Droid_Quick_Start_Guide/**
- Requirements that needs to be met: https://f-droid.org/docs/Inclusion_Policy/ (No Google Play, no proprietary tracking/reporting like Crashlytics or Firebase, ...

As and alternative (not preferred) you can maintain you own repository: https://f-droid.org/docs/Setup_an_F-Droid_App_Repo/

**Comment by prusnak** [comment]
We need Firebase for notifications, without them the whole app does not work.

**Comment by tomas-bartos1** [comment]
The solution for notification without Google is to run foreground service. This service then polls or keeps open websocket. This is what Signal, Telegram, Matrix and many more are doing.

- Android foreground service: https://developer.android.com/guide/components/foreground-services
- Matrix: https://github.com/vector-im/element-android/blob/develop/docs/notifications.md
- Telegram https://github.com/Telegram-FOSS-Team/Telegram-FOSS/blob/master/Notifications.md

Tradeoffs:
1. It keeps sticky notification that is forever there.
2. It may drain a bit more battery.
3. It may have delay (if its polling with long period to lower battery consumption)
4. It may require user to turn off battery optimization shit in Android

**Comment by tomas-bartos1** [comment]
There is also this solution https://unifiedpush.org/. But I don't have any experience with that nor did extensive research.

According to this it is used by some number of apps: https://unifiedpush.org/users/apps/

**Comment by tomas-bartos1** [comment]
I have just tested unifiedpush (installed [`ntfy`](https://unifiedpush.org/users/distributors/ntfy/)) and connected to Tusky and Element (Matrix client). It works. But not sure if it is useful for waking the app to do some work. I dont fully understand the underlying technical solution (how apps interacts)

**Comment by prusnak** [comment]
> But not sure if it is useful for waking the app to do some work.

Yes, we need to wake up the app to re-encrypt the existing offers for new people who just joined your network.
